src/features/broker-protection/extractors/address.js

```diff
@@ -0,0 +1,73 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { Extractor } from '../types.js'
+import { stringToList } from '../actions/extract.js'
+import parseAddress from 'parse-address'
+import { states } from '../comparisons/constants.js'
+
+/**
+ * @implements {Extractor<{city:string; state: string|null}[]>}
+ */
+export class CityStateExtractor {
+    /**
+     * @param {string[]} strs
+     * @param {import('../actions/extract.js').ExtractorParams} extractorParams
+     */
+    extract (strs, extractorParams) {
+        const cityStateList = strs.map(str => stringToList(str, extractorParams.separator)).flat()
+        return getCityStateCombos(cityStateList)
+    }
+}
+
+/**
+ * @implements {Extractor<{city:string; state: string|null}[]>}
+ */
+export class AddressFullExtractor {
+    /**
+     * @param {string[]} strs
+     * @param {import('../actions/extract.js').ExtractorParams} extractorParams
+     */
+    extract (strs, extractorParams) {
+        return strs
+            .map((str) => str.replace('\n', ' '))
+            .map((str) => stringToList(str, extractorParams.separator))
+            .flat()
+            .map((str) => parseAddress.parseLocation(str) || {})
+            // at least 'city' is required.
+            .filter((parsed) => Boolean(parsed?.city))
+            .map((addr) => {
+                return { city: addr.city, state: addr.state || null }
+            })
+    }
+}
+
+/**
+ * @param {string[]} inputList
+ * @return {{ city: string, state: string|null }[] }
+ */
+function getCityStateCombos (inputList) {
```

Comment:
This was just moved from the original place so that it can live alongside the extractor